src/nni_manager/training_service/pai/paiK8S/paiK8STrainingService.ts

```diff
@@ -144,66 +143,60 @@ class PAIK8STrainingService extends PAITrainingService {
             throw new Error('trial config is not initialized');
         }
         const jobName = `nni_exp_${this.experimentId}_trial_${trialJobId}`
-        const paiJobConfig: any = {
-            protocolVersion: 2, 
-            name: jobName,
-            type: 'job',
-            jobRetryCount: 0,
-            prerequisites: [
-              {
-                type: 'dockerimage',
-                uri: this.paiTrialConfig.image,
-                name: 'docker_image_0'
-              }
-            ],
-            taskRoles: {
-                taskrole: {
-                    instances: 1,
-                    completion: {
-                       minFailedInstances: 1,
-                       minSucceededInstances: -1
-                    },
-                    taskRetryCount: 0,
-                    dockerImage: 'docker_image_0',
-                    resourcePerInstance: {
-                        gpu: this.paiTrialConfig.gpuNum,
-                        cpu: this.paiTrialConfig.cpuNum,
-                        memoryMB: this.paiTrialConfig.memoryMB
-                    },
-                    commands: [
-                        command
-                    ]
-                }
-            },
-            extras: {
-                'com.microsoft.pai.runtimeplugin': [
-                    {
-                        plugin: this.paiTrialConfig.paiStoragePlugin
-                    }
+        let paiJobConfig: any = undefined;
+        if (this.paiTrialConfig.paiConfigPath) {
+                paiJobConfig = yaml.safeLoad(fs.readFileSync(this.paiTrialConfig.paiConfigPath, 'utf8'));
+                paiJobConfig.name = jobName;
+                paiJobConfig.taskRoles.taskrole.commands = [command]
```

Comment:
(I can't comment on L266 of this file.) Better to escape special characters in command. But it's not necessary for this PR.

Reply:
so there is only one key (i.e., `taskrole`) under `taskRoles`? what if there are multiple keys?

Reply:
NNI only parse the first taskrole, users should provide multiple taskroles in their configuration.

Reply:
you means only the first taskrole's command is wrapped by trial keeper? other taskroles' command is kept as it is

Reply:
Not the first taskrole, but the taskrole named "taskrole".

Reply:
update logic, NNI will use all of taskRoles now.